t/33-vagrant.t

```diff
@@ -0,0 +1,522 @@
+#!/usr/bin/perl
+
+use Cwd qw(abs_path);
+use Test::Most;
+use File::chdir;
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use Test::MockRandom 'backend::vagrant';
+use Test::MockObject;
+use Test::MockModule;
+use Test::Warnings ':report_warnings';
+use Test::Output qw(stderr_like);
+use backend::vagrant;
+
+use bmwqemu ();
+
+# hide debug messages, keep warnings & errors
+$bmwqemu::logger = Mojo::Log->new(level => 'warn');
+
+my $mock_vagrant = Test::MockModule->new("vagrant", no_auto => 1);
+$mock_vagrant->noop('mkdir');
+
+my $tmpdir        = File::Temp->newdir();
+my $mock_file_tmp = Test::MockModule->new("File::Temp");
+$mock_file_tmp->redefine('newdir', $tmpdir);
+
+my $expected_console_name        = undef;
+my $expected_console_type        = undef;
+my %expected_console_credentials = ();
+my $distri                       = Test::MockModule->new('distribution');
+$distri->redefine('add_console', sub {
+        my ($self, $name, $type, $creds) = @_;
+
+        if (defined($expected_console_name)) {
+            die "Invalid console name '$name'. Expected '$expected_console_name'" unless $name eq $expected_console_name;
+        }
+        if (defined($expected_console_type)) {
+            die "Invalid console type '$type'. Expected '$expected_console_type'" unless $type eq $expected_console_type;
+        }
+        if (%expected_console_credentials) {
+            is_deeply($creds, \%expected_console_credentials, 'Check the name of the console');
```

Comment:
here also I would have avoided the use of a Test::More function but nevermind

Reply:
Unfortunately I don't really know how to perform a deep comparison of hashes without relying on `is_deeply`.

Reply:
See my comment above for setting the test level

Reply:
Thanks! I've applied this to the current test code.